### DISCUSSION TITLE
When do we npm install to dependencies?

### DISCUSSION THREAD

**Discussion by manuganji** [sveltejs/kit#4469] [Q&A]
Is every package installed as a dev dependency with `npm i -D <package>` because svelte is compiled? In what cases will we do `npm i <package>`? 

EDIT: Sorry, I couldn't find a concrete answer from my reading. If this has been answered elsewhere, I'd like to know. I'm trying to understand compiled nature of svelte better.

**Comment by furudean** [comment]
**There is no meaningful heuristic between `devDependencies` and `dependencies` when it comes to Svelte Kit.**  Dependencies are tree-shaken at build time, meaning that modules are only included in your build bundle if they are actually used in your code. 

However I find it easier to visually grep what's used for what in my package.json by making anything that is used at runtime a `dependency` and anything else a `devDependency`.

**Reply by elliott-with-the-longest-name-on-github** [reply]
@furudean If I remember correctly, there's some special case of this with `adapter-node` (or there was at some point?). I specifically remember having to move some node-only dependencies from `devDependencies` to `dependencies` in order to be able to build. But it was like six or eight months ago, so it may have just been a bug.

**Comment by hyunbinseo** [comment]
```json
{
  "devDependencies":{
    "@sveltejs/adapter-auto":"workspace:*",
    "@sveltejs/kit":"workspace:*",
    "svelte":"^3.44.0",
    "svelte-preprocess":"^4.9.8",
    "typescript":"~4.6.2"
  },
  "dependencies":{
    "@fontsource/fira-mono":"^4.5.0",
    "@lukeed/uuid":"^2.0.0",
    "cookie":"^0.5.0"
  }
}
```
`create-svelte`'s [default template](https://github.com/sveltejs/kit/blob/2f5ec06be43eba7a0358259da2a802d70818c7a2/packages/create-svelte/templates/default/package.json) does distinguish `devDependencies` and `dependencies`.

**Comment by samlfair** [comment]
A colleague pointed out to me that:

> Since SvelteKit allows you to package your `lib` folder for reusability, you should install such "runtime" dependencies as `dependencies` so that they don't get bundled with the hypothetical package, but still installed alongside. This would allow projects relying on such package to benefit from updates matching the semver range specified for such dependencies without the need for a republish.

If you're publishing your app as a package, that's a good argument for installing packages as `dependencies`.
